### DISCUSSION TITLE
Unable to start pageserver using docker-compose

### DISCUSSION THREAD

**Discussion by william-lbn** [neondatabase/neon#8691] [General]
## Steps to reproduce
follow https://github.com/neondatabase/neon/blob/main/docs/docker.md  on mac pro

![Image](https://github.com/user-attachments/assets/fd84a1b0-a5fc-460c-8b8a-a92f0999c15a)

## Expected result

neon work it.

## Actual result

neon pageserver failed.

![Image](https://github.com/user-attachments/assets/dd0be541-7238-4350-9ee7-f2f6c4186e2f)

By checking docker-compose.yml, it was found that the pageserver had modified the startup method, but no entry for the startup command was found
﻿
![Image](https://github.com/user-attachments/assets/3c247dec-760a-4922-8a2d-64852066f680)

Add the startup command and enter the container to view

![Image](https://github.com/user-attachments/assets/10347cd0-067f-474b-90e9-7edc6cc351d6)

The file mounting is correct

![Image](https://github.com/user-attachments/assets/c5e2697d-509f-4dba-a65e-d039dff48cbf)

Pageserver is missing a startup command? or Is there no command to start the pageserver? 

![Image](https://github.com/user-attachments/assets/d559704a-2720-4e8b-83b2-b80af9acb75f)

What is the startup command for pageserver？

**Comment by problame** [comment]
Likely the new `identity.toml` file is missing in your existing data directory.

https://discord.com/channels/1176467419317940276/1184894814769127464/1266346201234804827

> Heads up self-hosted / neon_local users: we've moved the `id=` field of the pageserver configuration to a new configuration file `identity.toml`, which sits next to `pageserver.toml`.
> 
> To continue using your `neon_local` environment, make the following changes (assuming you are using `NEON_REPO_DIR=.neon`):
> * `cargo neon stop`
> * create new file `identity.toml`
> * move the `id=` line from `pageserver.toml`
> * `cargo neon start`
> 
> Here's the epic with more context in case you're interested: https://github.com/neondatabase/neon/issues/7736

**Comment by problame** [comment]
Also, please provide the pageserver log output.

**Comment by william-lbn** [comment]
Pageserver startup failed with no log output. 


![Image](https://github.com/user-attachments/assets/3591b467-8282-4def-9523-f501c3142b00)



![Image](https://github.com/user-attachments/assets/8f27c30a-d272-475f-8a94-c15416e29ec6)


![Image](https://github.com/user-attachments/assets/c9b5d8d0-9bea-46b4-8da7-7b225b3934e1)


![Image](https://github.com/user-attachments/assets/8f194358-8dc9-4227-a0d6-0e4fd8dd4a07)

I'm not sure where to go to check the logs, so I modified the image startup method and went to check. I found that the files were mounted correctly.

![Image](https://github.com/user-attachments/assets/a2318cea-96ab-44a2-94a2-4904a8406d77)

Do you know how to see the logs？


I see it has restarted dozens of times in a short period of time. It would be great if I could see the logs.

Or could you tell me the startup command for the page server so that I can try it out in the container.

**Comment by jcsp** [comment]
@william-lbn FYI, we recently added a README in `docker-compose/` to clarify the limited scope of the compose file:
>The configuration in this directory is used for testing Neon docker images: it is
not intended for deploying a usable system.  To run a development environment where
you can experiment with a minature Neon system, use `cargo neon` rather than container images.

There has been some discussion on the #neon-oss discord channel (https://discord.gg/92vNTzKDGp) about creating a kubernetes operator for running Neon services, which would provide a nicer way to manage compatibility of things like config files -- currently the neon binaries have backward compatibility for things like configs for about 2 weeks, as we release the code into our production environment at least weekly (but are open to working with the community on approaches to make things easier for self-hosted folks)
